src/generators.rs

```diff
@@ -74,27 +75,27 @@ impl Default for PedersenGens {
 /// orthogonal generators.  The sequence can be deterministically
 /// produced starting with an arbitrary point.
 struct GeneratorsChain {
-    reader: Sha3XofReader,
+    state: [u8; 32],
```

Comment:
nit: let's make `32` a constant for readability + reusability; also change other uses of the magic number